### PR TITLE
add train_test_split for 3%

### LINKED ISSUE #6: Séparation des données d'entraînement et test 3%

### PR DESCRIPTION
**Pull request by clervens**
this close #6

### CODE REVIEW

File: algorithms/cleaner.py [hunk 1]
```diff
@@ -31,3 +31,20 @@ def remove_duplicates(df):
     print("Second cut length:{}".format(len(df_no_dup_2)))
 
     return df_no_dup_2
+
+def train_test_split(df, test_size = .3, label_column = "quality"):
+    # Stratified sampling on the label column. In our case it's "quality"
+    grouped_df = df.groupby(label_column).apply(lambda x: x.sample(frac=test_frac))
```
Comment (komlan55) [inline algorithms/cleaner.py]: Je ne vois pas ou test_frac est declaré. Est-ce supposé être test_size??

Comment (clervens) [inline algorithms/cleaner.py]: Ouais, j'ai du l'oblier lorsque j'ai renommé mes variables

File: algorithms/cleaner.py [hunk 2]
```diff
@@ -31,3 +31,20 @@ def remove_duplicates(df):
...
+    
+    test_df = df.loc[index_in_test]
+    # The ~ sign reverse the booleans values
+    # Not sure if it works outside of pandas' dataframe
+    train_df = df.loc[~index_in_test]
+    
+    # return the values in the x_train, x_test, y_train, y_test format.
+    # Trying to replicate the format of 
+    # http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html
+    return train_df.values[:, :-1], test_df.values[:, :-1], train_df.values[:, -1], test_df.values[:, -1]
```
Comment (komlan55) [inline algorithms/cleaner.py]: Bon @clervens  dit moi si je comprend bien. On prend le fichier le Dataset.csv et on apres l'avoir normalisé, on sépare les donnés. On prend 70% pour l'apprentissage et 30% pour tester ce qu'on a appris a l'apprentissage right? DU moins cest supposer etre ca,mais ce que je ne comprend pas exactement cest pourquoi tu groupe les donnés par classes? On pourrait vivre sans non? Puis je ne comprend pas tout a fait ton slicing  dans le return. Pk tu retourne 4 array au lieu de simplement 2? Je comprend pas tu slice. Tu pourrais retourner seulement train_df et test_df non?

Comment (clervens) [inline algorithms/cleaner.py]: Groupe par classes pour avoir une bonne proportion des données.
Imaginons que dans notre ensemble de données contient 80 % de classe 1, 10 % de classe 2, 5 % de classes 3 et 5 % de classe 4.

En ne prenant que les 70 premiers pour cent du dataset, il est possible que l’ensemble d’entrainement ne contienne que des classes 1, donc toutes les prédictions donneront comme réponse classe 1.

En groupant les données, cela me permet de respecter la proportion des données en fonction de la colonne étiquette (quality).

pour le format de retour, je retourne les données sous le format `x_train, x_test, y_train, y_test` sous la forme d'un array (pas un array numy, ni un dataframe) pour permettre par la suite de les passer en paramètre dans la fonction d'entraiment `.fit(x_train, y_train)`

**Comment by nescobarification** [comment]
@clervens Ton code fonctionne bien! Est-ce que c'est lui que tu me disais qui te donnait des erreurs lors du training?
screen shot 2017-07-17 at 10 11 35 pm
